### ISSUE TITLE
some compiler warnings von Travis

### ISSUE THREAD

**Issue by guy-maurel** [uncrustify/uncrustify#2015]
The test:
https://travis-ci.org/uncrustify/uncrustify/jobs/424337308
is marked als passed, but there are still warnings, which should be removed.
```.txt
In file included from ../src/option.h:14:0,
                 from ../src/options.h:13,
                 from ../src/uncrustify_types.h:16,
                 from ../src/prototypes.h:11,
                 from ../src/backup.cpp:25:
../src/enum_flags.h: In instantiation of ‘uncrustify::flags<Enum>::flags(Enum) [with Enum = uncrustify::iarf_e]’:
../src/option.h:55:1:   required from here
../src/enum_flags.h:35:11: warning: parameter ‘flag’ set but not used [-Wunused-but-set-parameter]
    inline flags(Enum flag)
           ^
../src/enum_flags.h: In instantiation of ‘uncrustify::flags<Enum>::flags(Enum) [with Enum = uncrustify::tokenpos_e]’:
../src/option.h:85:1:   required from here
../src/enum_flags.h:35:11: warning: parameter ‘flag’ set but not used [-Wunused-but-set-parameter]
```

**Comment by mwoehlke-kitware** [comment]
Yuck... either that warning is impenetrably confusing, or it's bogus. First off, I don't see where the parameter is being "set". Second, it most certainly *is* used, to initialize `m_i`.

It is... interesting/informative that I don't see this with GCC 8.1.1. This makes me suspect that this is a bug in GCC 4.8.4.

**Comment by mwoehlke-kitware** [comment]
Yeah... it [goes away](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(j:1,lang:c%2B%2B,source:'template+%3Ctypename+Enum%3E+struct+flags%0A%7B%0A+++inline+flags(Enum+flag)%0A++++++:+m_i%7Bstatic_cast%3Cint%3E(flag)%7D+%7B%7D%0A+++int+m_i+%3D+0%3B%0A%7D%3B%0A%0Aenum+class+foo+%7B+bar+%7D%3B%0Atemplate+class+flags%3Cfoo%3E%3B%0A'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:g492,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',trim:'0'),lang:c%2B%2B,libs:!(),options:'-std%3Dc%2B%2B11+-Wall+-Wextra',source:1),l:'5',n:'0',o:'x86-64+gcc+4.9.2+(Editor+%231,+Compiler+%231)+C%2B%2B',t:'0')),k:50,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compiler:1,editor:1,wrap:'1'),l:'5',n:'0',o:'%231+with+x86-64+gcc+4.9.2',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:50,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4) in GCC 4.9.2. Definitely feels like a bug :cry:. In fact, it appears to be https://gcc.gnu.org/bugzilla/show_bug.cgi?id=59624 (note how the example is almost identical to our code), which matches up with the first GCC version in which the spurious warning is not produced.
